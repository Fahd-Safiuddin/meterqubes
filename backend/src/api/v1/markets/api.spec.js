const server = require('../../../index')
const agent = require('supertest').agent(server)
const token = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkIjp7ImlkIjoyLCJwdWJsaWNBZGRyZXNzIjoiMHhhOD' +
  'BmZjA4ZDA4ODA5YjI5YzczMzEzNWIxYjkzYTA3Nzk5ZTVkMTdmIn0sImlhdCI6MTU1NDI4NzA5NH0.u-tN-7YKjA3MzkTrQlDx5Bkd4' +
  '1lVRE9C8fs3_9e6KuM'

describe('Markets (API)', function () {
  afterAll(async () => {
    await server.close()
  })

  it('get all markets with token "BNB"', async () => {
    try {
      const response = await agent.get('/api/v1/markets?token=BNB')
        .catch(err => console.error(err))
      expect(response.status).toEqual(200)
      expect(response.body).toMatchSnapshot()
    } catch (e) {
      console.error(e)
      expect(e).toMatch('error')
    }
  })

  it('get all markets with token name "mk"', async () => {
    try {
      const response = await agent.get('/api/v1/markets?token=mk')
        .catch(err => console.error(err))
      expect(response.status).toEqual(200)
      expect(response.body).toMatchSnapshot()
    } catch (e) {
      console.error(e)
      expect(e).toMatch('error')
    }
  })

  it('get market with token Id=1', async () => {
    try {
      const response = await agent.get('/api/v1/markets/1')
        .set('Authorization', 'Bearer ' + token)
        .expect('Content-Type', /json/)
      expect(response.status).toEqual(200)
      expect(response.body).toMatchSnapshot()
    } catch (e) {
      console.error(e)
      expect(e).toMatch('error')
    }
  })
})
