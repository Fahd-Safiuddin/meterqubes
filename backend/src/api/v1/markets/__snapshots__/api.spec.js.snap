// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markets (API) get all markets with token "BNB" 1`] = `
Array [
  Object {
    "amountDecimals": 2,
    "asMakerFeeRate": "0.00100",
    "asTakerFeeRate": "0.00300",
    "baseToken": "BNB",
    "baseTokenAddress": "0xb8c77482e45f1f44de1745f52c74426c631bdd52",
    "baseTokenDecimals": 18,
    "baseTokenName": "Binance Coin",
    "baseTokenProjectUrl": "https://www.binance.com/",
    "id": 1,
    "marketOrderMaxSlippage": "0.10000",
    "minOrderSize": "1.000000000000000000",
    "priceDecimals": 8,
    "pricePrecision": 5,
    "quoteToken": "WETH",
    "quoteTokenAddress": "0x251bbfa0abf2dc356a44f7af5e4d7a224a4ec01f",
    "quoteTokenDecimals": 18,
    "supportedOrderTypes": Array [
      "LIMIT",
    ],
    "tokens": "BNB-WETH",
  },
]
`;

exports[`Markets (API) get all markets with token name "mk" 1`] = `
Array [
  Object {
    "amountDecimals": 2,
    "asMakerFeeRate": "0.00100",
    "asTakerFeeRate": "0.00300",
    "baseToken": "MKR",
    "baseTokenAddress": "0x9f8f72aa9304c8b593d555f12ef6589cc3a579a2",
    "baseTokenDecimals": 18,
    "baseTokenName": "Maker",
    "baseTokenProjectUrl": "https://makerdao.com/",
    "id": 2,
    "marketOrderMaxSlippage": "0.10000",
    "minOrderSize": "0.005000000000000000",
    "priceDecimals": 8,
    "pricePrecision": 5,
    "quoteToken": "WETH",
    "quoteTokenAddress": "0x251bbfa0abf2dc356a44f7af5e4d7a224a4ec01f",
    "quoteTokenDecimals": 18,
    "supportedOrderTypes": Array [
      "LIMIT",
    ],
    "tokens": "MKR-WETH",
  },
]
`;

exports[`Markets (API) get market with token Id=1 1`] = `
Object {
  "amountDecimals": 2,
  "asMakerFeeRate": "0.00100",
  "asTakerFeeRate": "0.00300",
  "baseToken": "BNB",
  "baseTokenAddress": "0xb8c77482e45f1f44de1745f52c74426c631bdd52",
  "baseTokenDecimals": 18,
  "baseTokenName": "Binance Coin",
  "baseTokenProjectUrl": "https://www.binance.com/",
  "id": 1,
  "marketOrderMaxSlippage": "0.10000",
  "minOrderSize": "1.000000000000000000",
  "priceDecimals": 8,
  "pricePrecision": 5,
  "quoteToken": "WETH",
  "quoteTokenAddress": "0x251bbfa0abf2dc356a44f7af5e4d7a224a4ec01f",
  "quoteTokenDecimals": 18,
  "supportedOrderTypes": Array [
    "LIMIT",
  ],
  "tokens": "BNB-WETH",
}
`;
